experimental/emulated_vault/opt/emulated-vault/adaptor/fs.py

```diff
@@ -0,0 +1,166 @@
+            return False
+        return True
+
+    def ping(self):
+        pingPath = os.path.join(self.basePath, self.pingRelPath)
+        success, value = self._get_param(pingPath)
+        if not success or value is None:
+            self.logger.error("no ping response")
+            return (False, "")
+        self.logger.debug("ping response: %s", value)
+        return (True, value)
+
+    def getParameter(self, parameterPath):
+        parameterPath = parameterPath.lstrip("/")
+        _parameterPath = os.path.join(self.basePath, parameterPath)
+        success, value = self._get_param(_parameterPath)
+        if not success:
+            self.logger.error("Failed to bring variable %s", parameterPath)
+            return False, ""
+        if value is None:
+            self.logger.error("Could not find variable %s", parameterPath)
+            return True, None
+        self.logger.debug("Parameter get response for %s ready", parameterPath)
+        #self.logger.debug("Parameter get response: %s", value)
+        return True, value
+
+    def searchParameters(self, parameterPathPrefix, keyFilters = {}, filters = {}):
+        parameterPathPrefix = parameterPathPrefix.lstrip("/")
+        _parametersPathPrefix = os.path.join(self.basePath, parameterPathPrefix)
+        success, items = self._get_params(_parametersPathPrefix, prefixToRemove=self.basePath, keyFilters=keyFilters)
+        if not success:
+            self.logger.error("Failed to bring variable by prefix %s", parameterPathPrefix)
+            return False, ""
+
+        filtered = {}
+        for key, val in items.iteritems():
+            skip = False
+            for filterName, filter in filters.iteritems():
+                if not filter(key, val):
+                    self.logger.debug("Parameter %s dropped, not matching filter %s", key, filterName)
+                    skip = True
+                    break
+            if skip:
+                continue
+            filtered[key] = val
+
+        return True, filtered
+
+
+    def setParameter(self, parameterPath, value):
+        parameterPath = parameterPath.lstrip("/")
+        _parameterPath = os.path.join(self.basePath, parameterPath)
+        success = self._set_param(_parameterPath, value)
+        if not success:
+            self.logger.error("Failed to set variable %s", parameterPath)
+            return False
+        return True
+
+    def deleteParameter(self, parameterPath):
+        parameterPath = parameterPath.lstrip("/")
+        _parameterPath = os.path.join(self.basePath, parameterPath)
+        success = self._delete_param(_parameterPath)
+        if not success:
+            self.logger.error("Failed to delete variable %s", parameterPath)
+            return False
+        return True
+
+    def _get_param(self, paramName):
+        self.logger.debug("Get parameter %s", paramName)
+        try:
+            with open(paramName) as fd:
+                value = fd.read()
+        except:
```

Comment:
doesn't logger.exception do just that?

Reply:
Not exactly. By "swallow errors" I don't mean that it hides them from ever being seen, just that it can capture more than you would want. A bare `except` can capture things like a `SystemExit`, `SystemError` that indicate an issue with the interpreter, or `KeyboardInterrupt` which indicates that a running user (or control system like `systemd`) wants it to stop.

`except`ing `Exception` is similarly frowned-upon in the general Python community, but would at least avoid those problems, because all of those classes inherit from `BaseException` not `Exception`. If this was me I'd probably write something like:

```python3
try:
    with open(paramName) as fd:
        value = fd.read()
except OSError:
    self.logger.exception("%s parameter not found.", paramName)
    return False, None
```
and then maybe something under that to handle `KeyboardInterrupt` without printing a stack trace in the logs:
```
except KeyboardInterrupt:
    logging.error("Exiting...")
    sys.exit(1)
```
or maybe not.